Tools/ardupilotwaf/chibios.py

```diff
@@ -53,8 +53,12 @@ class upload_fw(Task.Task):
     always_run = True
     def run(self):
         upload_tools = self.env.get_flat('UPLOAD_TOOLS')
+        upload_port = self.generator.bld.options.upload_port
         src = self.inputs[0]
-        return self.exec_command("{} '{}/uploader.py' '{}'".format(self.env.get_flat('PYTHON'), upload_tools, src))
+        if upload_port is None:
```

Comment:
Neater would be to append to a `command` string if there's an upload port.

Reply:
My python-foo is not great. Can you explain?

Reply:
Ok fixed